.github/workflows/pr.yml

```diff
@@ -10,13 +10,43 @@ on:
       - release/v[0-9]+.[0-9]+.[0-9]+
 
 jobs:
+  modified-files:
+    runs-on: ubuntu-latest
+    outputs:
+      app-changes: ${{ steps.filter.outputs.app }}
+      e2e-changes: ${{ steps.filter.outputs.e2e-pw }}
+      fiftyone-changes: ${{ steps.filter.outputs.fiftyone }}
+    steps:
+      - uses: actions/checkout@v4
+      - uses: dorny/paths-filter@v3
+        id: filter
+        with:
+          filters: |
+            app:
+              - 'app/**'
+            e2e-pw:
+              - 'e2e-pw/**'
```

Comment:
We'll want to run e2e tests when `app/**` or `fiftyone/**` change as well.

Reply:
that's handled down on like 43
> if: ${{ needs.modified-files.outputs.app-changes == 'true' || needs.modified-files.outputs.e2e-changes == 'true' || needs.modified-files.outputs.fiftyone-changes == 'true' }}

Reply:
Yes, exactly. It's handled a few lines below!